stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/repository/domain/message/internal/DatabaseMessageRepository.kt

```diff
@@ -170,6 +174,7 @@ internal class DatabaseMessageRepository(
      * @param message [Message]
      */
     override suspend fun deleteChannelMessage(message: Message) {
+        deletedMessageIds.add(message.id)
         messageCache.remove(message.id)
         scope.launchWithMutex(dbMutex) { messageDao.deleteMessage(message.cid, message.id) }
```

Comment:
Just out of curiosity, shouldn’t we verify that the DAO has successfully deleted the message before invalidating cache and adding the deleted If to the list? Given that the scope could be canceled or other issues might prevent the database from updating, it seems like this could lead to unsynced data.

Reply:
Hey @skydoves ,

It should be fine. We delete from cache first to prevent `false` reads from cache.
Event if cache is empty we will hit the DB.

Reply:
Deleted messages should not be requested by our SDKs, the only issue was on the insert process, because a deleted message was updated with a wrong states, but as far we included the message id of  the deleted message into the set of deleted messages, it is not updated anymore